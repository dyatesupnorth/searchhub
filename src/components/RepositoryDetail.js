import React from 'react'
const RepositoryDetail = props => (
    
    <div>
       <h2>Repository Detail here</h2>
    </div>
    
)


export default RepositoryDetail
