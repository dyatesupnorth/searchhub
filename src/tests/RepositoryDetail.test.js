import React from 'react'
import ReactDOM from 'react-dom'
import RepositoryDetail from '../components/RepositoryDetail'
import Enzyme, { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';
import github from './fixtures/github'


Enzyme.configure({ adapter: new Adapter() });


test('should render RepositoryDetail with a github repo', () => {
  const wrapper = shallow(<RepositoryDetail item={github.items[0]} />)
  expect(wrapper).toMatchSnapshot();

})

test('should render Results with empty message', () => {
  const wrapper = shallow(<RepositoryDetail item={{}} />)
  expect(wrapper).toMatchSnapshot();
  
})





