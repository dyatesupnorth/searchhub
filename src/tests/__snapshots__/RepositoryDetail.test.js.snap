// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render RepositoryDetail with a github repo 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryDetail
    item={
        Object {
            "archive_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/{archive_format}{/ref}",
            "archived": false,
            "assignees_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/assignees{/user}",
            "blobs_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/git/blobs{/sha}",
            "branches_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/branches{/branch}",
            "clone_url": "https://github.com/freeCodeCamp/freeCodeCamp.git",
            "collaborators_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/collaborators{/collaborator}",
            "comments_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/comments{/number}",
            "commits_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/commits{/sha}",
            "compare_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/compare/{base}...{head}",
            "contents_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/contents/{+path}",
            "contributors_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/contributors",
            "created_at": "2014-12-24T17:49:19Z",
            "default_branch": "staging",
            "deployments_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/deployments",
            "description": "The https://freeCodeCamp.org open source codebase and curriculum. Learn to code and help nonprofits.",
            "downloads_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/downloads",
            "events_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/events",
            "fork": false,
            "forks": 14052,
            "forks_count": 14052,
            "forks_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/forks",
            "full_name": "freeCodeCamp/freeCodeCamp",
            "git_commits_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/git/commits{/sha}",
            "git_refs_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/git/refs{/sha}",
            "git_tags_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/git/tags{/sha}",
            "git_url": "git://github.com/freeCodeCamp/freeCodeCamp.git",
            "has_downloads": true,
            "has_issues": true,
            "has_pages": false,
            "has_projects": true,
            "has_wiki": false,
            "homepage": "",
            "hooks_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/hooks",
            "html_url": "https://github.com/freeCodeCamp/freeCodeCamp",
            "id": 28457823,
            "issue_comment_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/issues/comments{/number}",
            "issue_events_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/issues/events{/number}",
            "issues_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/issues{/number}",
            "keys_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/keys{/key_id}",
            "labels_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/labels{/name}",
            "language": "JavaScript",
            "languages_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/languages",
            "license": Object {
              "key": "bsd-3-clause",
              "name": "BSD 3-Clause \\"New\\" or \\"Revised\\" License",
              "node_id": "MDc6TGljZW5zZTU=",
              "spdx_id": "BSD-3-Clause",
              "url": "https://api.github.com/licenses/bsd-3-clause",
            },
            "merges_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/merges",
            "milestones_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/milestones{/number}",
            "mirror_url": null,
            "name": "freeCodeCamp",
            "node_id": "MDEwOlJlcG9zaXRvcnkyODQ1NzgyMw==",
            "notifications_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/notifications{?since,all,participating}",
            "open_issues": 262,
            "open_issues_count": 262,
            "owner": Object {
              "avatar_url": "https://avatars0.githubusercontent.com/u/9892522?v=4",
              "events_url": "https://api.github.com/users/freeCodeCamp/events{/privacy}",
              "followers_url": "https://api.github.com/users/freeCodeCamp/followers",
              "following_url": "https://api.github.com/users/freeCodeCamp/following{/other_user}",
              "gists_url": "https://api.github.com/users/freeCodeCamp/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/freeCodeCamp",
              "id": 9892522,
              "login": "freeCodeCamp",
              "node_id": "MDEyOk9yZ2FuaXphdGlvbjk4OTI1MjI=",
              "organizations_url": "https://api.github.com/users/freeCodeCamp/orgs",
              "received_events_url": "https://api.github.com/users/freeCodeCamp/received_events",
              "repos_url": "https://api.github.com/users/freeCodeCamp/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/freeCodeCamp/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/freeCodeCamp/subscriptions",
              "type": "Organization",
              "url": "https://api.github.com/users/freeCodeCamp",
            },
            "private": false,
            "pulls_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/pulls{/number}",
            "pushed_at": "2018-07-02T12:16:34Z",
            "releases_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/releases{/id}",
            "score": 25.30111,
            "size": 35587,
            "ssh_url": "git@github.com:freeCodeCamp/freeCodeCamp.git",
            "stargazers_count": 292612,
            "stargazers_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/stargazers",
            "statuses_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/statuses/{sha}",
            "subscribers_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/subscribers",
            "subscription_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/subscription",
            "svn_url": "https://github.com/freeCodeCamp/freeCodeCamp",
            "tags_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/tags",
            "teams_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/teams",
            "trees_url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp/git/trees{/sha}",
            "updated_at": "2018-07-05T12:24:46Z",
            "url": "https://api.github.com/repos/freeCodeCamp/freeCodeCamp",
            "watchers": 292612,
            "watchers_count": 292612,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Repository Detail here
</h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Repository Detail here",
      },
      "ref": null,
      "rendered": "Repository Detail here",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Repository Detail here
</h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Repository Detail here",
        },
        "ref": null,
        "rendered": "Repository Detail here",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`should render Results with empty message 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryDetail
    item={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Repository Detail here
</h2>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Repository Detail here",
      },
      "ref": null,
      "rendered": "Repository Detail here",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Repository Detail here
</h2>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Repository Detail here",
        },
        "ref": null,
        "rendered": "Repository Detail here",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
